sparkctl/cmd/create.go

```diff
@@ -267,28 +272,120 @@ func isContainerLocalFile(file string) (bool, error) {
 	return false, nil
 }
 
+type Blob interface {
+	// TODO: With go-cloud supporting setting ACLs, remove implementations of interface
+	setPublicACL(ctx context.Context, bucket string, filePath string) error
+}
+
+type BlobHandler struct {
+	ctx    context.Context
+	bucket *blob.Bucket
+	blob   Blob
+}
+
+func (bh BlobHandler) uploadToBucket(
+	uploadScheme string,
+	uploadBucket string,
+	uploadPath string,
+	localFilePath string) (string, error) {
+	fileName := filepath.Base(localFilePath)
+	uploadFilePath := filepath.Join(uploadPath, fileName)
+	// Check if exists by trying to fetch metadata
+	reader, err := bh.bucket.NewRangeReader(bh.ctx, uploadFilePath, 0, 0)
+	defer reader.Close()
+	if (blob.IsNotExist(err)) || (err == nil && Override) {
+		fmt.Printf("uploading local file: %s\n", fileName)
+
+		// Prepare the file for upload.
+		data, err := ioutil.ReadFile(localFilePath)
+		if err != nil {
+			return "", fmt.Errorf("Failed to read file: %s", err)
+		}
+
+		// Open Bucket
+		w, err := bh.bucket.NewWriter(bh.ctx, uploadFilePath, nil)
+		if err != nil {
+			return "", fmt.Errorf("Failed to obtain bucket writer: %s", err)
+		}
+
+		// Write data to bucket and close bucket writer
+		_, writeErr := w.Write(data)
+		if err := w.Close(); err != nil {
+			return "", fmt.Errorf("Failed to close bucket writer: %s", err)
```

Comment:
We use lower case letter for the first word of the error string in this package.